### ISSUE TITLE
jsii releases are no longer from a trusted publisher

### ISSUE THREAD

**Issue by gabegorelick** [aws/jsii-compiler#2662]
https://github.com/aws/jsii-compiler/pull/2465, which was released with v5.9.18, configured trusted publishing. That seems to have been disabled in https://github.com/aws/jsii-compiler/pull/2626. Publishes are now done with a long-lived token (`NODE_AUTH_TOKEN`).

This means that consumers using pnpm with [trustPolicy: no-downgrade](https://pnpm.io/settings#trustpolicy) get an error when attempting to install recent versions. The unfortunate reality is that once a package is published with trusted publishing, all subsequent versions need it or else clients may complain about trust being downgraded.

> [ERR_PNPM_TRUST_DOWNGRADE] High-risk trust downgrade for "jsii@5.9.44" (possible package takeover)

See https://cdk-dev.slack.com/archives/C01C8R4K2E6/p1780675977611519?thread_ts=1779896079.602189&cid=C01C8R4K2E6

This issue is similar to #2651, but that was about downgrading provenance. That was fixed in #2652, but the trusted publishing downgrade is still present.

**Comment by mrgrain** [comment]
Thanks for the report. This is a false positive report, originating from a single package version that was published using trusted publishing. 

While trusted publishing is something we'd like to do, we are currently somewhat blocked by npmjs.com limitation of one trusted publisher config per package, while jsii is maintaining multiple version lines. This is also the reason we reverted from this single release back to tokens.

There are possible work arounds, but they require us to re-architect our GHA workflows. We will prioritize this in the wider context of our projects.

**Comment by mrgrain** [comment]
npm has said they might support multiple configs soon. Fingers crossed. 🤞🏻

**Comment by yohimik** [comment]
For tracking consumer impact while the multiple-publisher configuration limitation remains: I reproduced the direct-package rejection on current `jsii@5.9.53` and `6.0.13`, as well as the original `5.9.44`.

Each run used pnpm `11.15.0`, Node `v24.18.0`, and a separate workspace containing `pnpm-workspace.yaml` with `packages: []`. The exact command for the latest line was:

```sh
env CI=true pnpm add --ignore-scripts --trust-policy no-downgrade \
  --trust-policy-exclude @trickfilm400/rollup-plugin-off-main-thread \
  --trust-policy-exclude semver --save-exact jsii@6.0.13
```

The two exclusions isolate unrelated transitive-package rejections; `jsii` remains subject to the policy. All three runs exited 1 with `ERR_PNPM_TRUST_DOWNGRADE` naming the requested direct `jsii` version, followed by:

```text
Earlier versions had trusted publisher, but this version has provenance attestation.
```

This is consistent with the publisher-method change explained above, not evidence of compromised package contents. It makes the current affected versions and consumer configuration explicit. Once the publishing configuration changes, repeating this isolated check for each maintained line would verify that pnpm accepts the new release; registry availability or provenance alone would not catch this rejection.
